### DISCUSSION TITLE
`boot.mpy` and `main.mpy` are not loaded

### DISCUSSION THREAD

**Discussion by vshymanskyy** [micropython/micropython#15853] [General]
What's the reason behind MicroPython skipping to load `boot.mpy` and `main.mpy` automatically at start?
`.py` files are preferred over `.mpy` on regular imports, but the boot process seems to skip `.mpy` completely.

**Comment by shariltumin** [comment]
This is because the MicroPython main.c does these two things when powered up or reset:

1. ```int ret = pyexec_file_if_exists("boot.py");```
2. ```int ret = pyexec_file_if_exists("main.py");```

These will effectively do:

```py
>>> execfile("boot.py")
>>> execfile("main.py")
```

To use *.mpy we need to use ```import main``` or ```m=__import__("main")```.

**Reply by vshymanskyy** [reply]
Yes, but is there any reason why MicroPython doesn't do it by default?

**Reply by shariltumin** [reply]
I don't know why.

**Reply by shariltumin** [reply]
However, we have a simple workaround.

Rename your "main.mpy" to something else e.g. "Main.mpy" then in "main.py" you just need a line "import Main".

Maybe you already know this, but are not happy with the workaround.

**Reply by ricksorensen** [reply]
#10470

TBD
